flow/layers/layer_tree.cc

```diff
@@ -52,9 +54,22 @@ bool LayerTree::Preroll(CompositorContext::ScopedFrame& frame,
       device_pixel_ratio_};
 
   root_layer_->Preroll(&context, frame.root_surface_transformation());
+
+  RasterCache(&context, frame.root_surface_transformation());
+
   return context.surface_needs_readback;
 }
 
+void LayerTree::RasterCache(PrerollContext* context, const SkMatrix& ctm) {
+  for (unsigned i = 0; i < context->raster_cached_entries.size(); i++) {
+    auto& entry = context->raster_cached_entries[i];
+    if (entry.need_caching) {
+      entry.layer->TryToPrepareRasterCache(context, ctm);
```

Comment:
You are using the snapshotted context from the end of Preroll and the root surface CTM for caching. You need those values from the moment when the RasterCacheEntry was created. Perhaps look through the data used by "TryToPrepareRasterCache" and see if it needs the whole PrerollContext to do its work and just copy the values it needs into the RCE. If it is a lot of them, then you may need to copy the entire PrerollContext into the RCE on construction.